builds/e2e/e2e.yaml

```diff
@@ -212,6 +212,67 @@ jobs:
     timeoutInMinutes: 120
 
     steps:
+    - bash: |
+        set -e
+
+        # Install PowerShell dependencies
+        sudo apt-get update
+        sudo apt-get install -y liblttng-ust0 jq
+
+        # Get the PowerShell tarball
+        release_id=$(curl -sSH 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/powershell/powershell/releases/latest | jq '.id')
+        url=$(curl -sSH 'Accept: application/vnd.github.v3+json' https://api.github.com/repos/powershell/powershell/releases/$release_id/assets | jq -r ".[] | select(.name | contains(\"linux-arm64.tar.gz\")) | .browser_download_url")
+        echo "$release_id"
+        echo "$url"
+        file=$(basename $url)
+        curl -sSL $url -o /tmp/$file
+
+        # Install PowerShell
+        sudo mkdir -p /opt/microsoft/powershell/7
+        sudo tar zxf /tmp/$file -C /opt/microsoft/powershell/7
+        sudo chmod +x /opt/microsoft/powershell/7/pwsh
+        sudo ln -s /opt/microsoft/powershell/7/pwsh /usr/bin/pwsh
+      displayName: Install pwsh
```

Comment:
Please refactor `-bash:` into a template *.yaml/*.sh so we may further keep the e2e.yaml minimum.

Reply:
This is temporary and will be removed once 1ES arm64 pools support custom images. So I think it's better to keep it all in one place, where it looks ugly and every day we can't wait to remove it. ;)